pypsa/statistics.py

```diff
@@ -695,13 +809,19 @@ def supply(
         def func(n, c):
             sign = -1.0 if c in n.branch_components else n.df(c).get("sign", 1.0)
             ports = get_ports(n, c)
-            p = pd.DataFrame(0, index=n.snapshots, columns=n.df(c).index)
+            p = list()
             for port in ports:
                 mask = port_mask(n, c, port=port, bus_carrier=bus_carrier)
                 df = n.pnl(c)[f"p{port}"]
-                df = sign * mask * df.reindex(columns=p.columns, fill_value=0)
+                df = sign * df.reindex(columns=mask.index, fill_value=0)
```

Comment:
the mask is now taken only for reindexing, but not used for multiplication anymore. Is that intended?